src/components/OTable.vue

```diff
@@ -0,0 +1,245 @@
+
+            <slot name="controls-header__start-after" />
+          </slot>
+        </div>
+
+        <div class="o-table__controls__section o-table__controls__section-end">
+          <slot name="controls__header__end">
+            <slot name="controls-header__end-before" />
+
+
+            <template v-if="!hideSearch">
+              <slot name="controls-header__search">
+                <SearchInput v-model="search" :disabled="disableSearch" :placeholder="searchPlaceholder" :label="searchLabel" />
+              </slot>
+            </template>
+
+            <slot name="controls-header__end-after" />
+          </slot>
+        </div>
+      </slot>
+    </div>
+
+    <p-table v-bind="attrs" :data="_data" :columns="_columns">
+      <template #select-heading>
+        <p-checkbox v-if="!noData" v-model="select" :disabled="disableSelect" />
+        <span v-else />
+      </template>
+
+      <template #actions-heading>
+        <span />
+      </template>
+
+      <template #select="{ row }">
+        <p-checkbox v-model="_selectedRows" :value="row" :disabled="disableSelect" />
+      </template>
+
+      <template v-for="(_, name) in $slots" #[name]="slotData">
+        <slot :name="name" v-bind="slotData" />
+      </template>
+
+      <template #actions="{ row }">
+        <div class="o-table__row__actions">
+          <slot name="actions" :row="row" />
+        </div>
+      </template>
+
+
+      <template #empty-state>
+        <p-empty-results>
+          <template #message>
+            No {{ emptyResultsLabel }}
+          </template>
+          <template #actions>
+            <p-button v-if="!noData" size="sm" secondary @click="clear">
+              Clear Filters
+            </p-button>
+          </template>
+        </p-empty-results>
+      </template>
+    </p-table>
+
+    <div class="o-table__controls-footer">
+      <slot name="controls__footer">
+        <div class="o-table__controls__section o-table__controls__section-start">
+          <slot name="controls__footer__start" />
+        </div>
+
+        <div class="o-table__controls__section o-table__controls__section-end">
+          <slot name="controls__footer__end" />
+        </div>
+      </slot>
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+  export default {
+    name: 'OTable',
+    expose: [],
+    inheritAttrs: false,
+  }
+</script>
+
+<script lang="ts" setup>
+  import { TableColumn, TableData } from '@prefecthq/prefect-design'
+  import { ref, useAttrs, computed } from 'vue'
+  import { SearchInput, ResultsCount, SelectedCount } from '@/components'
+
+  export type TableDataFilter = (term: string, row: TableData, index: number, arr: TableData[]) => boolean
+
+  const attrs = useAttrs()
+
+  const props = withDefaults(defineProps<{
+    selectedRows?: TableData[],
+    columns: TableColumn[],
+    data: TableData[],
+    emptyResultsLabel?: string,
+    searchPlaceholder?: string,
+    searchLabel?: string,
+    rowsCountLabel?: string,
```

Comment:
Aren't these likely to be the same? "No things" "Search things" "10 things" etc

Reply:
Yeah agreed - I consolidated all of these down to a `label` prop except for the search placeholder, which is distinct